### PR TITLE
SDIT-1591 Add prisoner merge event handling for Sentencing Adjustment

### PR DESCRIPTION
**Pull request by andymarke**
This update includes essential changes in the SentencingAdjustmentsSynchronisationService to handle prisoner merge events so it can synchronize sentence and key date adjustments. Moreover, these modifications follow the introduction of SentencingAdjustmentsNomisApiService to interact with the NOMIS API. The update also involves new test cases ensuring that the prisoner merge handling works correctly.

### CODE REVIEW

File: helm_deploy/values-prod.yaml [hunk 1]
```diff
@@ -30,6 +30,7 @@ generic-service:
     FEATURE_EVENT_INCIDENT-DELETED-REQUIREMENTS: false
     FEATURE_EVENT_ALERT-UPDATED: false
     FEATURE_EVENT_ALERT-INSERTED: false
+    FEATURE_EVENT_BOOKING_NUMBER-CHANGED: false
```
Comment (andymarke) [inline helm_deploy/values-prod.yaml:34]: switch this off in prod until tested

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/sentencing/SentencingAdjustmentsNomisApiService.kt [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import org.springframework.web.reactive.function.client.WebClient
+import org.springframework.web.reactive.function.client.awaitBody
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.nomissync.model.SentencingAdjustmentsResponse
+
+@Service
+class SentencingAdjustmentsNomisApiService(@Qualifier("nomisApiWebClient") private val webClient: WebClient) {
+  suspend fun getAllByBookingId(bookingId: Long): SentencingAdjustmentsResponse = webClient.get()
+    .uri(
+      "/prisoners/booking-id/{bookingId}/sentencing-adjustments?active-only=false",
+      bookingId,
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/sentencing/SentencingAdjustmentsNomisApiService.kt:14]: actually active-only doesn't exist yet, but will add it next

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/sentencing/SentencingAdjustmentsNomisApiMockServer.kt [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo
+import com.github.tomakehurst.wiremock.matching.RequestPatternBuilder
+import org.springframework.http.HttpStatus
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.nomissync.model.KeyDateAdjustmentResponse
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.nomissync.model.SentenceAdjustmentResponse
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.nomissync.model.SentencingAdjustmentsResponse
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.wiremock.NomisApiExtension.Companion.nomisApi
+
+@Component
```
Comment (andymarke) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/sentencing/SentencingAdjustmentsNomisApiMockServer.kt:15]: I will move other NOMIS calls to this class in a separate PR

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/sentencing/SentencingPrisonOffenderEventListener.kt [hunk 4]
```diff
@@ -54,6 +54,10 @@ class SentencingPrisonOffenderEventListener(
                 (sqsMessage.Message.fromJson()),
               )
 
+              "BOOKING_NUMBER-CHANGED" -> sentencingAdjustmentsSynchronisationService.synchronisePrisonerMerge(
+                (sqsMessage.Message.fromJson()),
+              )
```
Comment (steverendell) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/sentencing/SentencingPrisonOffenderEventListener.kt:59]: interesting - all these params seem to have redundant brackets 🤷

**Review by steverendell — APPROVED** [review]
lgtm

**Pull request merged**